dask/cli.py

```diff
@@ -0,0 +1,91 @@
+import warnings
+
+from dask import __version__
+from dask.compatibility import entry_points
+
+try:
+    import click
+except ImportError as e:
+    msg = (
+        "The dask.cli module requires click to be installed.\n\n"
+        "Install with conda or pip:\n\n"
+        " conda install click\n"
+        " pip install click\n"
+    )
+    raise ImportError(msg) from e
+
+
+CONTEXT_SETTINGS = {
+    "help_option_names": ["-h", "--help"],
+    "max_content_width": 88,
+}
+
+
+@click.group(context_settings=CONTEXT_SETTINGS)
+@click.version_option(__version__)
+def cli():
+    """Dask command line interface."""
+    pass
+
+
+@cli.command()
+def docs():
+    """Open Dask documentation (https://docs.dask.org/) in a web browser."""
+    import webbrowser
+
+    webbrowser.open("https://docs.dask.org")
+
+
+@cli.group()
+def info():
+    """Information about your dask installation."""
+    pass
+
+
+@info.command()
+def versions():
+    """Print versions of Dask related projects."""
+    from dask.utils import show_versions
+
+    show_versions()
+
+
+def _register_command_ep(interface, entry_point):
+    """Add `entry_point` command to `interface`.
+
+    Parameters
+    ----------
+    interface : click.Command or click.Group
+        The click interface to augment with `entry_point`.
+    entry_point : importlib.metadata.EntryPoint
+        The entry point which loads to a ``click.Command`` or
+        ``click.Group`` instance to be added as a sub-command or
+        sub-group in `interface`.
+
+    """
+    command = entry_point.load()
```

Comment:
Just to note that this dynamic registration approach allows third party packages to register arbitrary code to be executed when the `dask` command is run.

Reply:
Thanks for raising this, it's a fair point.

I don't think we should worry about this though, I agree that it allows nasty people to register an entrypoint that would cause the `dask` CLI command to run arbitrary code. But they could equally set a `console_scripts` entrypoint that just overrides the `dask` command altogether. If multiple packages register the same CLI command they just get overridden, so any package who's name starts with a letter greater than `d` can override the `dask` command.

Entrypoints as a concept allows arbitrary packages to register code that will be executed by some other package. Typically we have to trust that everything installed in the Python environment is trusted. There isn't really any other option.